File: backend/src/main/java/it/unical/fleetgo/backend/Models/DTO/FatturaDaGenerareDTO.java
package it.unical.fleetgo.backend.Models.DTO;

import lombok.Getter;
import lombok.Setter;

@Setter
@Getter
public class FatturaDaGenerareDTO {
    Integer idAzienda;
    String nomeAzienda;
    String anno;
    String mese;
    Integer numeroNoleggi;
    Float costoTotale;
    Integer idOffertaApplicata;
    OffertaDTO offertaApplicata;

    public FatturaDaGenerareDTO() {}

    public FatturaDaGenerareDTO(Integer idAzienda,
                                String nomeAzienda,
                                String anno,
                                String mese,
                                Integer numeroNoleggi,
                                Float costoTotale) {
        this.idAzienda = idAzienda;
        this.nomeAzienda = nomeAzienda;
        this.anno = anno;
        this.mese = mese;
        this.numeroNoleggi = numeroNoleggi;
        this.costoTotale = costoTotale;
        this.idOffertaApplicata = 0;
        this.offertaApplicata = null;
    }
}

File: frontend/src/app/core/models/FatturaDaGenerareDTO.ts
import {OffertaDTO} from '@core/models/OffertaDTO';

export interface FatturaDaGenerareDTO {
  idAzienda: number;
  nomeAzienda: string;
  anno: string;
  mese: string;
  numeroNoleggi: number;
  costoTotale: number;
  idOffertaApplicata?: number;
  offertaApplicata?: OffertaDTO;
}
